tests/html5skin-test.js

```diff
@@ -0,0 +1,196 @@
+jest
+.dontMock('../js/skin')
+.dontMock('../js/components/utils')
+.dontMock('../js/components/accessibilityControls')
+.dontMock('../js/constants/constants')
+.dontMock('../config/skin.json')
+.dontMock('classnames');
+
+var $ = require('jquery');
+var _ = require('underscore');
+var sinon = require('sinon');
+var CONSTANTS = require('../js/constants/constants');
+
+var Html5Skin;
+
+// Mock OO environment needed for skin plugin initialization
+OO = {
+  playerParams: {
+    "core_version" : 4
+  },
+  publicApi: {},
+  EVENTS: {},
+  CONSTANTS: {
+    CLOSED_CAPTIONS: {}
+  },
+  VIDEO: {
+    ADS: 'ads',
+    MAIN: 'main'
+  },
+  init: function() {},
+  log: function() {},
+  plugin: function(module, callback) {
+    var plugin = callback(OO, _, $);
+    plugin.call(OO, OO.mb, 0);
+    // Save Html5Skin class in local var
+    Html5Skin = exposeStaticApi;
+  },
+  mb: {
+    subscribe: function() {},
+    unsubscribe: function() {},
+    publish: function() {},
+    addDependent: function() {}
+  }
+};
+// Requiring this will automatically initialize the plugin and set the Html5Skin var
+require('../js/controller');
+
+describe('Controller', function() {
+  var controller;
+
+  beforeEach(function() {
+    controller = new Html5Skin(OO.mb, 'id');
+    controller.state.pluginsElement = $('<div/>');
+    controller.state.pluginsClickElement = $('<div/>');
+    controller.state.mainVideoElement = {
+      classList: {
+        add: function() {},
+        remove: function() {}
+      },
+      getElementsByTagName: function() { return [controller.state.mainVideoElement] },
```

Comment:
How come we want this default behavior when calling getElementsByTagName?

Reply:
The idea was to return a mock dom element like `controller.state.mainVideoElement`, but it does look a bit confusing. I'll see if I can return something else here